src/js/tracing/reactnativetracing.ts

```diff
@@ -0,0 +1,221 @@
+      ...options,
+    };
+  }
+
+  /**
+   *  Registers routing and request instrumentation.
+   */
+  public setupOnce(
+    // @ts-ignore TODO
+    addGlobalEventProcessor: (callback: EventProcessor) => void,
+    getCurrentHub: () => Hub
+  ): void {
+    // eslint-disable-next-line @typescript-eslint/unbound-method
+    const {
+      traceFetch,
+      traceXHR,
+      tracingOrigins,
+      // @ts-ignore TODO
+      shouldCreateSpanForRequest,
+      routingInstrumentation,
+    } = this.options;
+
+    this._getCurrentHub = getCurrentHub;
+
+    routingInstrumentation?.registerRoutingInstrumentation(
+      this._onRouteWillChange.bind(this)
+    );
+
+    if (!routingInstrumentation) {
+      logger.log(
+        `[ReactNativeTracing] Not instrumenting route changes as routingInstrumentation has not been set.`
+      );
+    }
+
+    registerRequestInstrumentation({
+      traceFetch,
+      traceXHR,
+      tracingOrigins,
+      shouldCreateSpanForRequest,
+    });
+
+    addGlobalEventProcessor((event) => {
+      // eslint-disable-next-line no-empty
+      if (event.type === "transaction") {
+      }
+
+      return event;
+    });
+  }
+
+  /** To be called when the route changes, but BEFORE the components of the new route mount. */
+  private _onRouteWillChange(
+    context: TransactionContext
+  ): TransactionType | undefined {
+    // TODO: Consider more features on route change, one example is setting a tag of what route the user is on
+    return this._createRouteTransaction(context);
+  }
+
+  /** Create routing idle transaction. */
+  private _createRouteTransaction(
+    context: TransactionContext
+  ): TransactionType | undefined {
+    if (!this._getCurrentHub) {
+      logger.warn(
+        `[ReactNativeTracing] Did not create ${context.op} transaction because _getCurrentHub is invalid.`
+      );
+      return undefined;
+    }
+
+    // eslint-disable-next-line @typescript-eslint/unbound-method
+    const {
+      beforeNavigate,
+      idleTimeout,
+      maxTransactionDuration,
+    } = this.options;
+
+    const expandedContext = {
+      ...context,
+      trimEnd: true,
+    };
+
+    const modifiedContext =
+      typeof beforeNavigate === "function"
+        ? beforeNavigate(expandedContext)
+        : expandedContext;
+
+    if (modifiedContext.sampled === false) {
+      logger.log(
+        `[ReactNativeTracing] Will not send ${context.op} transaction.`
+      );
+    }
+
+    const hub = this._getCurrentHub();
+    const idleTransaction = startIdleTransaction(
+      hub as any,
+      context,
+      idleTimeout,
+      true
+    );
+    logger.log(
+      `[ReactNativeTracing] Starting ${context.op} transaction on scope`
+    );
+    idleTransaction.registerBeforeFinishCallback(
+      (transaction, endTimestamp) => {
+        adjustTransactionDuration(
+          secToMs(maxTransactionDuration),
+          transaction,
+          endTimestamp
+        );
+      }
```

Comment:
Here when calling `secToMs` inline, you can just do it in the `adjustTransactionDuration` internally. So you don't need to import `secToMs` here.